sky/engine/bindings/bindings.gni

```diff
@@ -48,3 +48,118 @@ idl_compiler_files = [
   "scripts/v8_types.py",
   "scripts/v8_utilities.py",
 ]
+
+template("dart_precompile") {
+  assert(defined(invoker.dart_package_root),
+         "The dart package root must be defined")
+  assert(defined(invoker.dart_script),
+         "The dart script must be specified")
+
+  vm_isolate_snapshot_name = target_name + "_vm_isolate.bin"
+  vm_isolate_snapshot = "$target_gen_dir/$vm_isolate_snapshot_name"
+  isolate_snapshot_name = target_name + "_isolate.bin"
+  isolate_snapshot = "$target_gen_dir/$isolate_snapshot_name"
+  
+  assembly_path =
+      "$target_gen_dir/" + target_name + "_instructions.S"
+  instructions_gen_target_name = target_name + "_instructions"
+  action(instructions_gen_target_name) {
+    deps = [
+      "//dart/runtime/bin:gen_snapshot($host_toolchain)",
+    ]
+    embedder_entry_points_manifest =
+        "//sky/engine/bindings/dart_vm_entry_points.txt"
+    inputs = [
+      "//dart/runtime/tools/create_snapshot_bin.py",
+      "//mojo/public/dart/mojo/sdk_ext/internal.dart",
+      "//mojo/public/dart/mojo/sdk_ext/src/handle_watcher.dart",
+      "//mojo/public/dart/mojo/sdk_ext/src/natives.dart",
+      "//mojo/public/dart/mojo/sdk_ext/src/timer_queue.dart",
```

Comment:
Yeah - I don't think we want to consider these filenames/paths API (same for sdk_ext/internal.dart below).

@johnmccutchan / @zanderso - any idea about how to package up knowledge of these parts of the mojo dart embedding so that flutter can reuse it sensibly?